main.go

```diff
@@ -442,7 +442,27 @@ func setupReconcilers(ctx context.Context, mgr ctrl.Manager) webhooks.ClusterCac
 		},
 	})
 	if err != nil {
-		setupLog.Error(err, "unable to create unstructured caching client")
+		setupLog.Error(err, "Unable to create unstructured caching client")
+		os.Exit(1)
+	}
+
+	// Setup a separate cache for the metadata watches to secrets.
+	// This way the watch does not use the LabelSelector defined at the cache which
+	// would filter to secrets having the cluster label, because secrets referred
+	// by ClusterResourceSet or ExtensionConfig are not specific to a single cluster.
```

Comment:
```suggestion
	// Setup a separate cache without label selector for secrets, to be used
	// when we need to watch for secrets that are not specific to a single cluster (e.g. ClusterResourceSet or ExtensionConfig controllers).
```

Reply:
I'm fine either way, but this suggestion doesn't seem to be done

Reply:
Should now be done.